### ISSUE TITLE
TF-Learn early stopping monitor doesn't disable dropout

### ISSUE THREAD

**Issue by waleedka** [tensorflow/tensorflow#2171]
I'm training a custom CNN model that uses dropout for regularization. I'm using tf-learn r0.8 (skflow)  ValidationMonitor as follows.

```Python
from tensorflow.contrib.learn.python.learn import monitors
m = monitors.ValidationMonitor(X_validate, Y_validate, 0, print_steps=100, 
    early_stopping_rounds=500)
regressor.fit(X_train, Y_train, monitor=monitor)
```

I noticed that after early stopping is triggered, the validation error I calculate on the final model is significantly different from what the ValidationMonitor reports as the best validation error. To narrow the problem I set `early_stopping_rounds = 1` but even then the difference between the "best" validation error and the one calculated on the final model was too large. Large enough to make the early stopping feature unusable for my case.

I tracked the issue down to the fact that ValidationMonitor calculates the validation error without disabling dropouts, which is why the results were so different from what you'd get from calling model.predict(). I implemented a fix here. 

I'd appreciate it if one of the maintainers reviews it and confirms that it's the right fix. If so, I'll create a pull request. 

https://github.com/waleedka/tensorflow/commit/ee095f83ffedf2e2cefff126b76191ab691d5097

**Comment by ian-plosker** [comment]
@waleedka I've noticed the same behavior. 👍  for tracking it down!

**Comment by aselle** [comment]
Closing since @ian-plosker found this is fixed in the 0.9 RC. The code has diverged significantly between the submitted diff and the current tflearn as well. Please reopen if 0.9 does not resolve this for you @waleedka. Thanks!
